app/dao/notifications_dao.py

```diff
@@ -210,6 +212,30 @@ def get_notifications_for_job(service_id, job_id, filter_dict=None, page=1, page
 
 
 @statsd(namespace="dao")
+def get_notification_billable_unit_count_per_month(service_id, year):
+    start, end = get_financial_year(year)
+
+    notifications = db.session.query(
+        NotificationHistory.created_at,
+        NotificationHistory.billable_units
+    ).order_by(
+        NotificationHistory.created_at
+    ).filter(
+        NotificationHistory.billable_units != 0,
+        NotificationHistory.service_id == service_id,
+        NotificationHistory.created_at >= start,
+        NotificationHistory.created_at < end
+    ).all()
```

Comment:
if you don't call `.all()` here it will stream it into the iterator below, which should be a bit more efficient